code/model/ErrorPage.php

```diff
@@ -274,47 +301,65 @@ public function fieldLabels($includerelations = true) {
 		
 		return $labels;
 	}
-	
+
 	/**
-	 * Returns an absolute filesystem path to a static error file
-	 * which is generated through {@link publish()}.
-	 * 
-	 * @param int $statusCode A HTTP Statuscode, mostly 404 or 500
-	 * @param String $locale A locale, e.g. 'de_DE' (Optional)
+	 * Get error content for the given status code. Only returns pre-cached content
+	 * stored either in the asset store or in the static path (if enabled).
 	 *
+	 * @param int $statusCode A HTTP Statuscode, typically 404 or 500
 	 * @return string
 	 */
-	public static function get_filepath_for_errorcode($statusCode, $locale = null) {
-		if (singleton('ErrorPage')->hasMethod('alternateFilepathForErrorcode')) {
-			return singleton('ErrorPage')-> alternateFilepathForErrorcode($statusCode, $locale);
+	public static function get_content_for_errorcode($statusCode) {
+		// Attempt to retrieve content from generated file handler
+		$filename = self::get_error_filename($statusCode);
+		$storeFilename = File::join_paths(self::config()->store_filepath, $filename);
+		$result = self::get_asset_handler()->getGeneratedContent($storeFilename, 0);
+		if($result) {
+			return $result;
 		}
 
-		if(class_exists('Translatable') && singleton('SiteTree')->hasExtension('Translatable') && $locale && $locale != Translatable::default_locale()) {
-			return self::config()->static_filepath . "/error-{$statusCode}-{$locale}.html";
-		} else {
-			return self::config()->static_filepath . "/error-{$statusCode}.html";
+		// Fallback to physical store
+		if(self::config()->enable_static_file) {
+			$staticPath = self::config()->static_filepath . "/" . $filename;
+			if(file_exists($staticPath)) {
+				return file_get_contents($staticPath);
+			}
 		}
 	}
-	
+
 	/**
-	 * Set the path where static error files are saved through {@link publish()}.
-	 * Defaults to /assets.
+	 * Gets the filename identifier for the given error code.
+	 * Used when handling responses under error conditions.
 	 *
-	 * @deprecated 4.0 Use "ErrorPage.static_file_path" instead
-	 * @param string $path
+	 * @param int $statusCode A HTTP Statuscode, typically 404 or 500
+	 * @param ErrorPage $instance Optional instance to use for name generation
+	 * @return string
 	 */
-	static public function set_static_filepath($path) {
-		Deprecation::notice('4.0', 'Use "ErrorPage.static_file_path" instead');
-		self::config()->static_filepath = $path;
+	protected static function get_error_filename($statusCode, $instance = null) {
+		if(!$instance) {
+			$instance = ErrorPage::singleton();
+		}
+		// Allow modules to extend this filename (e.g. for multi-domain, translatable)
+		$name = "error-{$statusCode}.html";
+		$instance->extend('updateErrorFilename', $name, $statusCode);
```

Comment:
We should add a github issue to subsites once this is merged to adapt the new API, could easily be missed otherwise (https://github.com/search?q=alternateFilepathForErrorcode&type=Code&utf8=%E2%9C%93)